### ISSUE TITLE
Console Does Not Work with Python 3.9 64-bit version

### ISSUE THREAD

**Issue by t-coulter** [py2exe/py2exe#80]
I was working with Py2exe with Python 3.9

When I created my setup.py file, I used the console option as opposed to the windows option, as I wanted to create a console output. However, a console window would never open up when I tried this in Python 3.9. I even tried the example "Hello World.py" and no output to the console when I generated the executable.

I migrated to Python 3.7 as I saw some forums said to try version 3.7 and the console option now works. Please look into why the console option does not work for Python 3.9.

**Comment by Athanasius** [comment]
```
> python --version
Python 3.9.2
> type hello.py
print("Hello World")
> type setup.py
from distutils.core import setup
import py2exe

setup(console=[{"script": "hello.py"}],
      options={"py2exe": {
               "verbose": 4}
              },
     )
> python setup.py py2exe
running py2exe

  3 missing Modules
  ------------------
? _posixshmem                         imported from multiprocessing.resource_tracker, multiprocessing.shared_memory
? readline                            imported from cmd, code, pdb
? resource                            imported from test.support
Building 'dist\hello.exe'.
Add resource PYTHONSCRIPT/1(2393 bytes) to dist\hello.exe
Building shared code archive 'dist\library.zip'.
Add Loader for _queue.pyd to dist\library.zip
Add Loader for _ctypes.pyd to dist\library.zip
Add Loader for _elementtree.pyd to dist\library.zip
Add Loader for _decimal.pyd to dist\library.zip
Add Loader for _overlapped.pyd to dist\library.zip
Add Loader for _multiprocessing.pyd to dist\library.zip
Add Loader for _testcapi.pyd to dist\library.zip
Add Loader for _asyncio.pyd to dist\library.zip
Add Loader for select.pyd to dist\library.zip
Add Loader for _tkinter.pyd to dist\library.zip
Add Loader for _testinternalcapi.pyd to dist\library.zip
Add Loader for _ssl.pyd to dist\library.zip
Add Loader for _bz2.pyd to dist\library.zip
Add Loader for _lzma.pyd to dist\library.zip
Add Loader for _hashlib.pyd to dist\library.zip
Add Loader for unicodedata.pyd to dist\library.zip
Add Loader for pyexpat.pyd to dist\library.zip
Add Loader for _socket.pyd to dist\library.zip
Copy c:\program files (x86)\python39-32\python39.dll to dist
Copy C:\Program Files (x86)\Python39-32\DLLs\unicodedata.pyd to dist\unicodedata.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_bz2.pyd to dist\_bz2.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_lzma.pyd to dist\_lzma.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\select.pyd to dist\select.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\pyexpat.pyd to dist\pyexpat.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_socket.pyd to dist\_socket.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_hashlib.pyd to dist\_hashlib.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_decimal.pyd to dist\_decimal.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_ssl.pyd to dist\_ssl.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_queue.pyd to dist\_queue.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_elementtree.pyd to dist\_elementtree.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_ctypes.pyd to dist\_ctypes.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_tkinter.pyd to dist\_tkinter.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_testinternalcapi.pyd to dist\_testinternalcapi.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_testcapi.pyd to dist\_testcapi.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_asyncio.pyd to dist\_asyncio.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_multiprocessing.pyd to dist\_multiprocessing.pyd
Copy C:\Program Files (x86)\Python39-32\DLLs\_overlapped.pyd to dist\_overlapped.pyd
Copy DLL C:\Program Files (x86)\Python39-32\DLLs\libcrypto-1_1.dll to dist\
Copy DLL C:\Program Files (x86)\Python39-32\DLLs\tcl86t.dll to dist\
Copy DLL C:\Program Files (x86)\Python39-32\DLLs\libssl-1_1.dll to dist\
Copy DLL C:\Program Files (x86)\Python39-32\DLLs\libffi-7.dll to dist\
Copy DLL C:\Program Files (x86)\Python39-32\DLLs\tk86t.dll to dist\
> cd dist
> hello.exe
Hello World
```
This is Windows 10 Home, up to date with March 2021 patches.  I use the python.org version of Python (32-bit as implied by the install path).

```
> pip show py2exe
Name: py2exe
Version: 0.10.2.1
Summary: Build standalone executables for Windows
Home-page: http://www.py2exe.org/
Author: Thomas Heller
Author-email: theller@ctypes.org
License: MIT/X11
Location: c:\users\athan\appdata\roaming\python\python39\site-packages
Requires: cachetools, pefile
Required-by:
```

**Comment by t-coulter** [comment]
I created Python3 to run Python 3.9 and Python to run Python 3.7 then ran both codes:
`> python3 --version
Python 3.9.1
> python --version
Python 3.7.0
> type hello.py
print("Hello world!")
> type setup.py
from distutils.core import setup
import py2exe
 
setup(console=[{"script": "hello.py"}],
      options={"py2exe": {
               "verbose": 4}
              },
     )
> python setup.py py2exe
running py2exe
 
  1 missing Modules
  ------------------
? readline                            imported from cmd, code, pdb
Building 'dist\hello.exe'.
Add resource PYTHONSCRIPT/1(1917 bytes) to dist\hello.exe
Building shared code archive 'dist\library.zip'.
Add Loader for unicodedata.pyd to dist\library.zip
Add Loader for _socket.pyd to dist\library.zip
Add empty __init__ for implicit namespace package builtins to dist\library.zip
Add empty __init__ for implicit namespace package sys to dist\library.zip
Add Loader for select.pyd to dist\library.zip
Add Loader for _lzma.pyd to dist\library.zip
Add empty __init__ for implicit namespace package _imp to dist\library.zip
Add empty __init__ for implicit namespace package _frozen_importlib to dist\library.zip
Add Loader for _ctypes.pyd to dist\library.zip
Add Loader for _bz2.pyd to dist\library.zip
Add Loader for _hashlib.pyd to dist\library.zip
Add Loader for _ssl.pyd to dist\library.zip
Add Loader for pyexpat.pyd to dist\library.zip
Copy c:\users\timothy.coulter\appdata\local\programs\python\python37\python37.dll to dist
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python37\DLLs\select.pyd to dist\select.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python37\DLLs\pyexpat.pyd to dist\pyexpat.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python37\DLLs\unicodedata.pyd to dist\unicodedata.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python37\DLLs\_socket.pyd to dist\_socket.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python37\DLLs\_bz2.pyd to dist\_bz2.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python37\DLLs\_lzma.pyd to dist\_lzma.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python37\DLLs\_hashlib.pyd to dist\_hashlib.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python37\DLLs\_ssl.pyd to dist\_ssl.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python37\DLLs\_ctypes.pyd to dist\_ctypes.pyd
Copy DLL C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python37\DLLs\libcrypto-1_1-x64.dll to dist\
Copy DLL C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python37\DLLs\libssl-1_1-x64.dll to dist\
> .\hello.exe
Hello world!
> python3 setup.py py2exe
running py2exe
 
  2 missing Modules
  ------------------
? _posixshmem                         imported from multiprocessing.resource_tracker, multiprocessing.shared_memory
? resource                            imported from test.support
Building 'dist\hello.exe'.
Add resource PYTHONSCRIPT/1(2394 bytes) to dist\hello.exe
Building shared code archive 'dist\library.zip'.
Add Loader for unicodedata.pyd to dist\library.zip
Add Loader for select.pyd to dist\library.zip
Add Loader for win32api.pyd to dist\library.zip
Add Loader for _testcapi.pyd to dist\library.zip
Add Loader for _bz2.pyd to dist\library.zip
Add Loader for _ctypes.pyd to dist\library.zip
Add Loader for _ssl.pyd to dist\library.zip
Add Loader for _overlapped.pyd to dist\library.zip
Add Loader for _socket.pyd to dist\library.zip
Add Loader for pyexpat.pyd to dist\library.zip
Add Loader for _tkinter.pyd to dist\library.zip
Add Loader for _lzma.pyd to dist\library.zip
Add Loader for _win32sysloader.pyd to dist\library.zip
Add Loader for _queue.pyd to dist\library.zip
Add Loader for _hashlib.pyd to dist\library.zip
Add Loader for _asyncio.pyd to dist\library.zip
Add Loader for _elementtree.pyd to dist\library.zip
Add Loader for _decimal.pyd to dist\library.zip
Add Loader for win32evtlog.pyd to dist\library.zip
Add Loader for _multiprocessing.pyd to dist\library.zip
Add Loader for _testinternalcapi.pyd to dist\library.zip
Copy c:\users\timothy.coulter\appdata\local\programs\python\python39\python39.dll to dist
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\select.pyd to dist\select.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_bz2.pyd to dist\_bz2.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_lzma.pyd to dist\_lzma.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\unicodedata.pyd to dist\unicodedata.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_socket.pyd to dist\_socket.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_queue.pyd to dist\_queue.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_hashlib.pyd to dist\_hashlib.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_decimal.pyd to dist\_decimal.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_ssl.pyd to dist\_ssl.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\lib\site-packages\win32\_win32sysloader.pyd to dist\_win32sysloader.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\lib\site-packages\win32\win32api.pyd to dist\win32api.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\lib\site-packages\win32\win32evtlog.pyd to dist\win32evtlog.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_ctypes.pyd to dist\_ctypes.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\pyexpat.pyd to dist\pyexpat.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_elementtree.pyd to dist\_elementtree.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_tkinter.pyd to dist\_tkinter.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_testinternalcapi.pyd to dist\_testinternalcapi.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_testcapi.pyd to dist\_testcapi.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_asyncio.pyd to dist\_asyncio.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_multiprocessing.pyd to dist\_multiprocessing.pyd
Copy C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\_overlapped.pyd to dist\_overlapped.pyd
Copy DLL C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\libffi-7.dll to dist\
Copy DLL C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\tk86t.dll to dist\
Copy DLL C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\libssl-1_1.dll to dist\
Copy DLL C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\tcl86t.dll to dist\
Copy DLL C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\DLLs\libcrypto-1_1.dll to dist\
Copy ExtensionDLL C:\Users\timothy.coulter\AppData\Local\Programs\Python\Python39\Lib\site-packages\pywin32_system32\pywintypes39.dll to dist\
> .\hello.exe
>
`
You can see that there are different results from each, with the Py2exe code using the Console option.

**Comment by Athanasius** [comment]
Is yours 64-bit python ?  That would be one difference.

**Comment by t-coulter** [comment]
> Is yours 64-bit python ? That would be one difference.

Yes mine is 64-bit Python.

**Comment by albertosottile** [comment]
Hi, 

Unfortunately, I am not able to reproduce your problem as the script runs fine in my local test environment. Furthermore, as you can see by looking at the `setup.py` scripts [here](https://github.com/py2exe/py2exe/tree/master/tests/functional), all the CI tests use the `console` mode and they are pass in all the supported platforms, including 3.9 x64.

Hence, I think this is a problem unique to your environment. To further debug this, it would be helpful to get runtime logs from the executable.

So, could you run the packed `hello.exe` executable in an environment with `PY2EXE_VERBOSE` set to 2, as in
```
python setup.py py2exe
set PY2EXE_VERBOSE=2
dist\hello.exe
```
and report here the resulting log? (The log can be quite large so, it might be better to put that in a gist or in a pastebin or something like that). Thanks.

**Comment by t-coulter** [comment]
> Hi,
> 
> Unfortunately, I am not able to reproduce your problem as the script runs fine in my local test environment. Furthermore, as you can see by looking at the `setup.py` scripts [here](https://github.com/py2exe/py2exe/tree/master/tests/functional), all the CI tests use the `console` mode and they are pass in all the supported platforms, including 3.9 x64.
> 
> Hence, I think this is a problem unique to your environment. To further debug this, it would be helpful to get runtime logs from the executable.
> 
> So, could you run the packed `hello.exe` executable in an environment with `PY2EXE_VERBOSE` set to 2, as in
> 
> ```
> python setup.py py2exe
> set PY2EXE_VERBOSE=2
> dist\hello.exe
> ```
> 
> and report here the resulting log? (The log can be quite large so, it might be better to put that in a gist or in a pastebin or something like that). Thanks.

I ran the set PY2EXE_VERBOSE=2 command and I did not see any output. I have not used this command before so I do not know if I am using the PY2EXE_VERBOSE correctly.

What I do see as a main difference between my Python 3.7 version run and Python 3.9 version was the fact that there were different missing modules listed.

Python 3.7 shows the readline module is missing.

Python 3.9 shows the _posixshmem and resource modules are missing. Are those required to run or expected to be missing? This might be what is causing the problem.

My application is now working with Python 3.7, but it took a while to understand why the application did not work in Python 3.9.

Thank you.

**Comment by albertosottile** [comment]
Missing modules are expected behavior, as you can read in the CI logs. Perhaps more puzzling is the fact that in your 3.9 also modules like `pywin32` are embedded in the app and they should not. 

Just to confirm that you set the environment variable well, could you try to set it again and run the `hello.exe` executable packed with 3.7? This should generate the log.

**Comment by albertosottile** [comment]
Closing this as it is not related to the codebase but rather to the specific environment. Feel free to open another issue if you experience further problems with `py2exe`.
